# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/libdsk/hotfix-major.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

/usr/lib64/gcc/x86_64-t2-linux-gnu/15.1.0/../../../../x86_64-t2-linux-gnu/bin/ld: ../lib/.libs/libdsk.so: undefined reference to `major'
/usr/lib64/gcc/x86_64-t2-linux-gnu/15.1.0/../../../../x86_64-t2-linux-gnu/bin/ld: ../lib/.libs/libdsk.so: undefine
d reference to `major'

--- libdsk-1.4.2/lib/compress.c.vanilla	2025-06-26 15:38:35.879016648 +0200
+++ libdsk-1.4.2/lib/compress.c	2025-06-26 15:38:51.129183617 +0200
@@ -28,6 +28,8 @@
 #include <sys/stat.h>
 #endif
 
+#include <sys/sysmacros.h>
+
 #define TMPDIR "/tmp"
 
 static COMPRESS_CLASS *classes[] =
--- libdsk-1.4.2/lib/drvlinux.c.vanilla	2025-06-26 15:38:55.466809913 +0200
+++ libdsk-1.4.2/lib/drvlinux.c	2025-06-26 15:39:01.473973744 +0200
@@ -45,7 +45,7 @@
 #include <unistd.h>
 #endif
 
-
+#include <sys/sysmacros.h>
 
 /* This struct contains function pointers to the driver's functions, and the
  * size of its DSK_DRIVER subclass */
